akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala

```diff
@@ -278,6 +267,11 @@ abstract class EventSourcedBehavior[Command, Event, State] private[akka] (
 
 }
 
+@FunctionalInterface
+trait ReplicationInterceptor[Event, State] {
+  def intercept(originReplica: ReplicaId, sequenceNumber: Long, state: State, event: Event): CompletionStage[Done]
```

Comment:
maybe not important, but to align with event handler it might look better to have state and event parameters first?

Reply:
Changed order and introduced a SAM for Scala as well with some docs, makes it much less cryptict: [acedf0b](https://github.com/akka/akka/pull/32526/commits/acedf0bc23fb83dd09443a3218e9b2b01d9c1607)